validation/README.md

```diff
@@ -0,0 +1,116 @@
+5.  *language* - The implementation process should support different programming languages if required. For example, the Excel templates will include data validation where possible. However, not all validation steps can be included in Excel, and data validation occurs in multiple steps. Therefore, all validation steps are also able to be executed in Python.
+6.  *description* - Each validation step should include a clear human-readable description in English and French.
+7.  *required* - Validation should indicate whether a variable is required.
+8.  *dependencies* - Whether there are dependencies on other variables or validation steps, in addition to the order of validation.
+9.  *warning* - The number and type of warnings should be presented in human-readable form. Warnings should be presented in different formats, platforms, programs, etc.
+10. *errors* - The number and type of errors should be presented in human-readable form. Errors should be presented in different formats, platforms, programs, etc.
+11. *normalization (transformation*) - if values can be transformed, then this should occur. For example, data submitted with a non-SI unit (e.g., liters/hour) could be converted to its SI equivalent (m3/s) Variable mapping should be considered during variable mapping and not during validation, despite recognizing that mapping could be performed using normalization methods.
+12. *coercion* - If values can be coerced, then this should happen. i.e. integer as text should be coerced to numbers and a warning issued that this occurred.
+13. *version-specific* - Validation should indicate what version of the ODM the step is applicable for. Provisions for coercion and/or transforming old versions variables, types, and coercion of values to the most recent version. Warnings and errors should be issued, as appropriate.
+14. *wide table* - Validation of the wide table should be supported. Derived variable names should be validated, if possible.
+15. *extensible* - Wastewater-based surveillance is rapidly developing. The validation process should be customizable and extensible for new processes not identified in this list of validation processes.
+
+## Validation schema and rules
+
+A validation schema is a set of rules or validation dictionaries. The ODM will create the main schema that is updated for each version. However, the schema method can be used by ODM users to extend or develop schemas for their specific applications. ODM will provide a means for sharing schema, similar to the means for sharing input templates. The list of validation rules below is drawn from the Cerberus Python validation library. Cerberus package will be used.
+
+By defining a validation schema for the ODM, we will be able to create and maintain a consistent universal data entry format.
```

Comment:
A lot of the items below would benefit from having examples on how they would be implemented in the validation schema.